### ISSUE TITLE
Turn off logging

### ISSUE THREAD

**Issue by excitement-engineer** [oleksiyp/mockk#37]
Thanks for this project. Really great stuff!!

I was wondering whether it is possible to turn off logging? I am getting all kinds of output such as `INFO` messages in the console.

Thanks in advance

**Comment by tomkoptel** [comment]
👍 +1  I am looking for the same option. What about?

```kotlin
val dependency = mockk<Dependency>(logging = false)
```

**Comment by oleksiyp** [comment]
It would complicate things I believe. If I would just do `info` messages
`debug` would that solve the issue?

2018-03-06 17:18 GMT+01:00 tomkoptel <notifications@github.com>:

**Comment by tomkoptel** [comment]
What do you propose? To disable `info` and `debug` logging? I don't think it is a good idea to remove logs. I would still prefer to see the option to enable/disable logs.

**Comment by oleksiyp** [comment]
I propose to not have `info` logs, so by default nothing is shown. There is just one `info` message saying what is an answer. On the other hand you always can disable/show logging in `logback.xml`(any other SLF4J framework) anyway. Doing it per mock is not an option because it would need to revisit each logging message and somehow link it to this flag.

**Comment by tomkoptel** [comment]
It sounds like it is required to have `SLF4J ` on the classpath and then to include `logback.xml` in the resources folder. Is it correct?

**Comment by oleksiyp** [comment]
Yes, `logback-test.xml` even better

**Comment by excitement-engineer** [comment]
I like the idea of not showing any logs by default. Frameworks like mockito also don't do this for example. I think that the logs are only useful when debugging the code so being able to easily turn this on/off would be a win I think.

**Comment by oleksiyp** [comment]
The idea was to keep one top-level info message. It can help to see what has answered actually in your tests by mocking library. But I think this easily can be switched to DEBUG

**Comment by tomkoptel** [comment]
Me personally not against the logging it was a good idea, but when you put additional logs in the source code to see sequence call with your tests it becomes a little bit noisy from the mock instances. I would still prefer to have logs, but with an option to disable them on demand. To use `SLF4J` sounds like a solution, but it requires to add yet another lib just for the sake of disabling logs, which sounds like an overkill.

**Comment by oleksiyp** [comment]
Version 1.7.10 released, please check if by default has no logging and close the ticket.

**Comment by excitement-engineer** [comment]
Awesome thanks so much, it works:) Really appreciate it!

**Comment by novaterata** [comment]
@oleksiyp I'm confused by this. I don't want mockk logging and I use slf4j myself, but my logback is handled by dropwizard, so I don't have a logback.xml or test-logback.xml. Do I still need to create and configure a test-logback.xml to disable logging. It makes finding build failures in gradle output frustrating because there are all these mock DEBUG statements I have to wade through

**Comment by oleksiyp** [comment]
Unfortunately, DEBUG is what MockK cannot leave without. I usually request to turn on DEBUG logging to capture what is happening on the other side. It is pretty weird that it is turned on by default for tests with Dropwizard. So you need to configure `test-logback.xml` to fix it.

I can think of a switch in MockK properties, but you will still need to create some file with settings, so that is almost the same as creation of `test-logback.xml`

**Comment by novaterata** [comment]
I figured it out, thanks, I don't know why it was Debug by default

**Comment by AlexCzar** [comment]
I tried adding test-logback.xml (and logback-test.xml) with following content:
```xml
<configuration>
	<root level="warn"/>
</configuration>
```
This doesn't seem to affect logging in any way, my log is still full of `[Test worker] DEBUG io.mockk.impl...` lines.
How do I disable this?

**Comment by laxa88** [comment]
For those curious:

Mockk uses [Logback](https://logback.qos.ch/).

You need to add an `logback-test.xml` or `logback.xml` file in the `resource` folder. Any other filename won't be recognised.
<img width="294" alt="Screen Shot 2022-03-01 at 11 33 24" src="https://user-images.githubusercontent.com/668915/156093760-3cdfa70a-bd1d-4c7a-bd85-a7803309356c.png">

Then, in the xml, add something like this:
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration debug="true">
    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
            <pattern>%msg%n</pattern>
        </encoder>
    </appender>
    <root level="debug">
        <appender-ref ref="STDOUT" />
    </root>
</configuration>
```
Notes:
- Change the value to `level="info"` or `"warn"` as you see fit. 
- You need to define an `<appender>` first, then use it in the `<appender-ref>`.
- You need to define the message pattern in `<pattern>`
- Note the the `ref` can be any arbitrary value instead of `STDOUT`.

Some links for reference:
- https://www.sentinelone.com/blog/getting-started-quickly-kotlin-logging/
- https://logback.qos.ch/manual/configuration.html

**Comment by erksch** [comment]
After multiple hours I am still unable to turn off the DEBUG logs of mockk in my multiplatform project.

I enabled standard streams for my test logging:

```
testLogging {
        showStandardStreams = true
}
```

I tried to add the `logback.xml` or `logback-test.xml` to `shared/src/commonTest/resources/`.
I also tried to only enable standard streams for INFO log level

```
testLogging {
       info { 
             showStandardStreams = true
      }
}
``` 

But nothing seems to work. I want logging for all my INFO level logs but not the DEBUG level logs from mockk.
Has anyone found a solution, or could explain how to verify that the logback setup works?
